### ISSUE TITLE
CI: lint is failing because of output to stderr

### ISSUE THREAD

**Issue by andyfaff** [scipy/scipy#13179]
The lint on CI is currently [failing](https://dev.azure.com/scipy-org/SciPy/_build/results?buildId=9396&view=logs&jobId=3a559e2a-952e-58d2-b8db-2e604a9266d7&j=3a559e2a-952e-58d2-b8db-2e604a9266d7&t=0dc7cd25-c0cc-5e9c-2964-e6eb48aa3daf) because a `pip install` step is designed to fail if anything writes to stderr.

```
  - script: >-
      python -m pip install
      pycodestyle==2.5.0
      pyflakes==2.1.1
    displayName: 'Install tools'
    failOnStderr: true
```
However, pip is emitting a "You are using pip version 20.2.4; however, version 20.3 is available", which this step treats as an error.
I think we should remove the `failOnStderr` line.

@tylerjereddy

**Comment by tylerjereddy** [comment]
Fixed by gh-13181

**Comment by andyfaff** [comment]
Sorry, just figured it was an easy fix. Closed in #13181.
If the pip install does fail, then the next part of the run will fail because pycodestyle won't be installed.
